devices/Ssd13xx/Ssd13xx.cs

```diff
@@ -42,13 +44,42 @@ public abstract class Ssd13xx : IDisposable
         /// </summary>
         public IFont Font { get; set; }
 
+        private GpioController _gpioController;        
+        private int _resetPin;
+        private bool _shouldDispose;
+        private bool _disposed = false;
+
         /// <summary>
         /// Constructs instance of Ssd13xx
         /// </summary>
         /// <param name="i2cDevice">I2C device used to communicate with the device</param>
+        /// <param name="resetPin">Reset pin (some displays might be wired to share the microcontroller's
+        /// reset pin).</param>
         /// <param name="resolution">Screen resolution to use for device init.</param>
-        public Ssd13xx(I2cDevice i2cDevice, DisplayResolution resolution = DisplayResolution.OLED128x64)
+        /// <param name="gpio">Gpio Controller.</param>
+        /// <param name="shouldDispose">True to dispose the GpioController.</param>
+
+        public Ssd13xx(
+            I2cDevice i2cDevice,
+            DisplayResolution resolution = DisplayResolution.None,
+            int resetPin = -1,            
+            GpioController gpio = null,
+            bool shouldDispose = true)
         {
+
+            if (resolution == DisplayResolution.None) resolution = DisplayResolution.OLED128x64;            
+
+            _resetPin = resetPin;            
+            if (resetPin >= 0)
+                {
+                _gpioController = gpio ?? new GpioController();
+                if (!_gpioController.IsPinModeSupported(_resetPin, PinMode.Output))
+                        throw new ArgumentException($"GPIO pin {_resetPin} must support output mode.");                
+                this.Reset();
+                }
+
+            _shouldDispose = shouldDispose;
```

Comment:
it's more complex than that. If the GpioController was null, then you should dispose it regardless of what is said. So:
```suggestion
            _shouldDispose = shouldDispose || (gpio == null);
```